genie_registry/vcf.py

```diff
@@ -215,3 +172,97 @@ def _validate(self, vcfdf):
         # will parse a VCF in a detailed way,
         # and output with warnings or errors if the format is not adhered too
         return total_error, warning
+
+    def validate_tumor_and_normal_sample_columns(self, input_df: pd.DataFrame) -> str:
+        """
+        Validates that the expected tumor sample column and optional normal
+        sample columns are present in the VCF depending on how many
+        columns you have present in the VCF and they have no missing values
+
+        Rules:
+            - VCFs can only have a max of 11 columns including the 9 required columns
+            - For 11 columns VCFs, it is assumed this is a matched tumor normal vcf file
+                which means there should be a tumor sample and normal sample
+                column present
+            - For 10 column VCFs, it is assumed this is a single sample vcf file
+                which means there should be a tumor sample column present
+            - Anything lower than 10 columns is INVALID because you must have at
+            least a tumor sample column on top of the 9 required VCF columns
+            - If tumor sample and/or normal sample columns are present, they must not have
+            any missing values.
+
+        Example: VCF with Matched Tumor Normal columns
+            | OTHER_VCF_COLUMNS | GENIE-GOLD-1-1-tumor | GENIE-GOLD-1-1-normal |
+            | ----------------- | -------------------- | --------------------- |
+            |        ...        |         ...          |          ...          |
+
+        Example: VCF with Single Tumor VCF column
```

Comment:
Do you mean single sample vcf here?

Reply:
Yeah actually the nomenclature can be both ways VCF with single sample column or VCF with single tumor column (single sample), I'll make both examples